text/0034-hash-field-bom.md

```diff
@@ -0,0 +1,72 @@
+# Update Hash Field in Bill of Materials
+
+## Summary
+
+Currently, the only hashing algorithm that we support in the BOM (bill of
+materials) is `SHA256`, as the only checksum fields are explicitly named
+`sha256`. This shoehorns the hash algorithm to `SHA256` an alienates any hashes
+we can collect in other formats. This RFC proposes that the Paketo BOM supports
+hashes in any algorithm.
+
+## Motivation
+
+In the original BOM RFC it was proposed to just have fields named `sha256` as
+that was the only hashing algorithm used by dependencies hosted by Paketo.
+However, in the pursuit of enhancing our BOM by adding information about things
+such as node module information, we have started to use third party tools to
+help generate the BOM. These third party tools return industry-standard BOM
+formats that include a wide and varying array of hashes using varying hashing
+algorithms. This has led us to conclude that our BOM specification needs to
+update to be inclusive of a wide range of hashing algorithms to people that are
+not hosting the packages they use and the metadata attached.
+
+## Implementation
+
+We purpose the following change to the BOM schema:
+```
+[[bom]]
+name = "<dependency name>"
+
+[bom.metadata]
+  arch = "<compatible processor architecture>"
+  cpe = "<version-specific common platform enumeration>"
+  deprecation-date = "<dependency EOS date formatted in using RFC 3339>"
+  licenses = [<dependency license IDs(s) in SPDX Format>]
+  purl = "<package URL per github.com/package-url>"
+  summary = "<package summary>"
+  uri = "<compiled dependency URI>"
+  version = "<dependency version>"
+
+[bom.metadata.checksum]
```

Comment:
Should we support multiple checksums? CycloneDX seems to specify that a single package might have multiple checksum values in their [spec documentation](https://cyclonedx.org/use-cases/#packaging-and-distribution).

Reply:
Again this is something I thought about but it would just be a representation of hashes for the same artifact using different algorithms if my understanding is correct. That kind of structure just feels kind of like overkill at the current moment and muddies the schema. I would be willing to go back and revise this at a later date though.